core/src/test/java/org/wso2/msf4j/util/UtilsTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.msf4j.util;
+
+import org.testng.annotations.Test;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * Tests for utility class.
+ */
+public class UtilsTest {
+
+    @Test
+    public void testNormalizePath() {
+        assertEquals("hello", Utils.normalizePath("hello"));
+        assertEquals("hello", Utils.normalizePath("/hello"));
+        assertEquals("hello", Utils.normalizePath("hello/"));
+        assertEquals("hello", Utils.normalizePath("/hello/"));
+        assertEquals("hello", Utils.normalizePath("//hello"));
+        assertEquals("hello", Utils.normalizePath("///hello"));
+        assertEquals("hello", Utils.normalizePath("///hello///"));
+
+        assertEquals("hel/lo", Utils.normalizePath("hel/lo"));
+        assertEquals("hel/lo", Utils.normalizePath("hel//lo"));
+        assertEquals("hel/lo", Utils.normalizePath("hel///lo"));
+        assertEquals("hel/lo", Utils.normalizePath("//hel//lo//"));
+
+        assertEquals("h/e/ll/o", Utils.normalizePath("h/e/ll/o"));
+        assertEquals("h/e/ll/o", Utils.normalizePath("h//e//ll//o"));
+        assertEquals("h/e/ll/o", Utils.normalizePath("//h//e//ll//o//"));
+
+        assertEquals("", Utils.normalizePath(""));
+        assertEquals("", Utils.normalizePath("/"));
+        assertEquals("", Utils.normalizePath("//"));
+        assertEquals("", Utils.normalizePath("///"));
+        assertEquals("h", Utils.normalizePath("h"));
+        assertEquals("h", Utils.normalizePath("/h"));
+        assertEquals("h", Utils.normalizePath("h/"));
```

Comment:
I think we have already tested below 2 assertions in line 30 and 31. Hence the below 2 assertions are not required
```java
    assertEquals("h", Utils.normalizePath("/h"));
    assertEquals("h", Utils.normalizePath("h/"));
```

Reply:
I like to treat these tests as "black box" tests and thus trying to test edge conditions here like empty path, path only consisting one or more slashes, or path of length 1.  Of course with the current implementation, there is no chance that these will fail. But my original optimized code did array index ops and for that off-by-one errors could have been possible.

I can add some more comments to the test cases explaining if that helps.

Reply:
Yes. You are right. A comment similar to "Testing the edge case of the path of length 1" would help for the above 2 assertions :)